monitoring/rid_qualifier/flight_data_generator.py

```diff
@@ -0,0 +1,479 @@
+
+
+class TrackWriter():
+
+    """
+        Write the tracks created by AdjacentCircularFlightsSimulator into disk (in the outputs directory) as GeoJSON FeatureCollection 
+        Args:
+        flight_path_points: A set of flight path points generated by generate_flight_paths_points method in the AdjacentCircularFlightsSimulator class
+        bboxes: A set of bounding boxes generated by generate_query_bboxes method in the AdjacentCircularFlightsSimulator class
+        country_code: An ISO 3166-1 alpha-3 code for a country
+
+        Outputs: 
+        GeoJSON files for bboxes created in the `test_definitions/{country_code}` folder 
+
+
+    """
+
+    def __init__(self, grid_tracks: List[GridCellFlight], bboxes: List[QueryBoundingBox], country_code='CHE') -> None:
+        ''' This class uses the same output directory as the AdjacentCircularFlightsSimulator class and requires the path points (Tracks) and the bounding boxes from that class.
+
+        '''
+
+        self.grid_cells_flight_tracks = grid_tracks
+        self.bboxes = bboxes
+        self.country_code = country_code
+        self.output_directory = Path('test_definitions', self.country_code)
+        # Create test_definition directory if it does not exist
+        self.output_directory.mkdir(parents=True, exist_ok=True)
+        self.output_subdirectories = (Path(self.output_directory, 'tracks'), Path(self.output_directory, 'query_bboxes'))
+
+        for output_subdirectory in self.output_subdirectories:
+            output_subdirectory.mkdir(parents=True, exist_ok=True)
+
+    def write_bboxes(self):
+        ''' This module writes the bboxes as a GeoJSON FeatureCollection '''
+        for buffered_bbox_details in self.bboxes:
+
+            features = json.dumps({'type': 'Feature', 'properties': {"timestamp_before": buffered_bbox_details.timestamp_before.isoformat(), "timestamp_after": buffered_bbox_details.timestamp_after.isoformat()}, 'geometry': shapely.geometry.mapping(buffered_bbox_details.shape)})
+            bbox_file_name = 'box_%s.geojson' % buffered_bbox_details.name
+            
+            bbox_output_path = self.output_subdirectories[1] / bbox_file_name
+
+            with open(bbox_output_path, 'w') as f:
+                f.write(features)
+
+    def write_tracks(self):
+        ''' This module writes tracks as a GeoJSON FeatureCollection (of Point Feature) for use in other software '''
+
+        flight_point_current_index = {}
+        num_flights = len(self.grid_cells_flight_tracks)
+        for i in range(num_flights):
+            flight_point_current_index[i] = 0
+
+        for track_id, grid_cell_flight_track in enumerate(self.grid_cells_flight_tracks):
+            feature_collection = {"type": "FeatureCollection", "features": []}
+            point_collection = []
+            for cur_track_point in grid_cell_flight_track.track:
+                p = Point((cur_track_point.lat, cur_track_point.lng, cur_track_point.alt))
```

Comment:
`shapely.geometry.point.Point` accepts arguments (x, y, z), so this is passing `cur_track_point.lat` as x and `cur_track_point.lng` as y.  This puts the GeoJSON for CHE in Ethiopia.  I recommend viewing the output GeoJSON as a sanity check as I expect that may reveal at least one additional issue not mentioned specifically in this review.

Reply:
Yes agreed, the co-ordinates are fixed now, I am looking into the other issue.

Reply:
As of now, I'm still seeing the additional issue (should be apparent from a qualitative GeoJSON visualization).

Reply:
Yes, the generated track is a oval and not a circle, this is a projection issue, fixing this now.

Reply:
OK this is now fixed and I have the first draft of the changes.